### PR TITLE
example: fix casual/example/work

### LINKED ISSUE #349: casual/example/work does nothing
When calling casual/example/work it is expected to take at least the specified amount of time, preferably doing som cpu intensive work for the duration of the call

Currently it will exit immediately, regardless of configured duration.

from logfile:
```
<domain>|e8e3e8275713454f91dfa8dbbe713fce|9278|139780810491840|casual-example-server||||information|example server started with arguments: [--work, 1s]
<domain>|e8e3e8275713454f91dfa8dbbe713fce|9278|139780810491840|casual-example-server||||information|sleep: 0ns
<domain>|e8e3e8275713454f91dfa8dbbe713fce|9278|139780810491840|casual-example-server||||information|work: 1000000000ns

```

make call and check statistics:

```
%> casual service --metric-reset
%> echo '{ "key" : "some", "value": "json"}' | casual buffer --compose ".json/" | casual call --service casual/example/work | casual buffer --extract
{ "key" : "some", "value": "json"}

%>casual service -ls | grep work
casual/example/work                                          example   D  join    90.000000            1  1  0.000046  0.000046  0.000046  0  0.000000   0   0  2024-03-19T11:32:34.156331+01:00
```

### PR DESCRIPTION
**Pull request by partickl**
Correct the busyloop so we actually wait for the intended duration
Add simple unittest

Resolves #349

### CODE REVIEW

**Comment by lazan** [comment]
Should this be done with 1.6? Hmm, it would be practical regarding test-cases I presume...

File: middleware/example/server/source/example/work.cpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ namespace casual
          {
             auto deadline = platform::time::clock::type::now() + local::global.work;
 
-            while( deadline < platform::time::clock::type::now())
+            while( deadline > platform::time::clock::type::now())
```
Comment (lazan) [inline middleware/example/server/source/example/work.cpp:105]: Good catch! 😄

File: middleware/example/server/unittest/source/test_example.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+      TEST( example_server, work_500ms__expect_at_least_500ms_elapsed_time)
+      {
+         common::unittest::Trace trace;
+         
+         auto domain = local::domain();
+
+         auto elapsed_time = local::timed_action(local::call_work);
+
+         EXPECT_TRUE( elapsed_time > std::chrono::milliseconds{ 500});
```
Comment (lazan) [inline middleware/example/server/unittest/source/test_example.cpp]: I would like to see a shorter duration. If we've got 10 of these it's 5s. 10ms should be sufficient?

Comment (lazan) [reply middleware/example/server/unittest/source/test_example.cpp]: ... that is, we don't want to end up running all unittests takes several minutes.

Comment (partickl) [reply middleware/example/server/unittest/source/test_example.cpp]: agreed, i'll shorten as much as possible.

**Review by lazan — APPROVED** [review]

**Pull request merged**